### ISSUE TITLE
'no-unused-vars' reports vars assigned with underscore arrow function as error

### ISSUE THREAD

**Issue by carlosvigil** [eslint/eslint#8681]
**Tell us about your environment**

* **ESLint Version:**3.19
* **Node Version:**8
* **npm Version:**5.0.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default
**Please show your full configuration:**
Also, occurs in AirBnB base, not in Standard
```
{
    "env": {
        "browser": true,
        "es6": true
    },
    "extends": "eslint:recommended",
    "parserOptions": {
        "sourceType": "module"
    },
    "rules": {
        "indent": [
            "error",
            2
        ],
        "linebreak-style": [
            "error",
            "unix"
        ],
        "quotes": [
            "error",
            "single"
        ],
        "semi": [
            "error",
            "never"
        ]
    }
}
```

**What did you do? Please include the actual source code causing the issue.**
assigned an arrow using underscore

```js

 request.onerror = _ => reject(Error('Error getting data.'))

```

**What did you expect to happen?**
Able to use underscore instead of empty parentheses.

**What actually happened? Please include the actual, raw output from ESLint.**
[eslint] '_' is defined but never used. (no-unused-vars)
(parameter) _: any

**Comment by not-an-aardvark** [comment]
Thanks for the report. However, this is working as indented -- `_` is a parameter which is not used.

**Comment by not-an-aardvark** [comment]
Closing because this is working as intended.

**Comment by ljharb** [comment]
@carlosvigil you want `() => reject(…)` - in other words, `_` is a real argument, `()` means "no arguments".

**Comment by carlosvigil** [comment]
Thanks!
